api.php

```diff
@@ -0,0 +1,211 @@
+	return $response;
+});
+
+/********* USER ROUTES *********/
+
+// lists a users info including their reviews //
+$app->get('/user/{id}', function (Request $request, Response $response, array $args) {
+	$id = $args['id'];
+	$user = User::findUserById($id);
+	if(empty($user)){
+		$response->getBody()->write('user not found');
+		return $response->withStatus(404);
+	}
+	//returning users info
+	$response->getBody()->write(json_encode($user->jsonSerialize()));
+	return $response;
+});
+
+// Create new user
+$app->post('/users/new', function (Request $request, Response $response, array $args) {
+	$body = json_decode($request->getBody());
+
+	$user = new User($body->first_name, $body->last_name, $body->email, $body->password, $body->phone, $body->preferred ?? false);
+	$response->getBody()->write(json_encode($user));
+	return $response
+		->withHeader('Content-Type', 'application/json')
+		->withStatus(201);
+});
+
+//Change info for a user that already exists
+$app->post('/user/{id}', function (Request $request, Response $response, array $args) {
+	$id = $args['id'];
+	//Locate id
+	$user = User::findUserById(intval($id));
+	$body = json_decode($request->getBody());
+
+	//Update user info
+	if (isset($body->first_name)) {
+		$user->setFirstName($body->first_name);
+	}
+
+	if (isset($body->last_name)) {
+		$user->setLastName($body->last_name);
+	}
+
+	if (isset($body->email)) {
+		$user->setEmail($body->email);
+	}
+
+	if (isset($body->password)) {
+		$user->setPassword($body->password);
+	}
+
+	if (isset($body->phone)) {
+		$user->setPhoneNumber($body->phone);
+	}
+	var_dump($user);
+
+	$response->getBody()->write(json_encode($user));
+	return $response
+		->withHeader('Content-Type', 'application/json')
+		->withStatus(201);
+});
+
+
+/********* REVIEW ROUTES *********/
+
+// /reviews - displays all reviews
+$app->get('/reviews', function (Request $request, Response $response, array $args) {
+	// get our sort/filter vals
+	$params = $request->getQueryParams();
+	$filterVal = (isset($params['filter-by']) ? $params['filter-by'].' '.$params['filter-val'] : '');
+	$reviews = Review::getAllReviews($filterVal, $params['archived'] ?? false);
+	//return review info based on ID
+	$response->getBody()->write(json_encode($reviews));
+	return $response
+		->withHeader('Content-Type', 'application/json')
+		->withStatus(200);
+});
+
+// /review/{reviewID} - displays the info about a specific review not just the review contents
+$app->get('/review/{reviewID}', function (Request $request, Response $response, array $args) {
+	$reviewID = intval($args['reviewID']);
+	$review = \Review::getReviewByID($reviewID);
+	//return review info based on ID
+	$response->getBody()->write(json_encode($review));
+	return $response
+		->withHeader('Content-Type', 'application/json')
+		->withStatus(200);
+});
+
+//review/new  Creates a new review
+$app->post('/review/new', function (Request $request, Response $response, array $args) {
```

Comment:
I am getting a "Script:Error: Unexpected token u in JSON at position 0" error message and I believe it has to do with the new Review__constructor setting the UserID as Null.

Reply:
I need to see what you sent.
Please post the request EXACTLY as you sent it.

Reply:
ok 
in the URL {{URL}}/review/new


```
{
    "score": 4,
    "comment": "this is a test comment for new review teeee3st"
}
```

then got back 

{
    "score": 4,
    "comment": "this is a test comment for new review teeee3st"
}

Reply:
ok i cleaned up your comment a bit but im not seeing the error...

Reply:
interesting, we can go over it in a few minutes, I'm probably using postman wrong.

Reply:
@margeschrec are you still having this issue?

Reply:
@billythakidd04 yes